cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/StageContext.java

```diff
@@ -118,4 +119,23 @@
    */
   @Nullable
   Schema getOutputSchema();
+
+  /**
+   * Discover the base URL for a Service, relative to which Service endpoints can be accessed
+   *
+   * @param applicationId Application name
+   * @param serviceId     Service name
+   * @return URL for the discovered service or null if the service is not found
+   */
+  @Nullable
+  URL getServiceURL(String applicationId, String serviceId);
```

Comment:
I guess for now we can extend `ServiceDiscoverer` and later if we do not want methods from `ServiceDiscoverer`, we can always copy methods. I will make this change